### PR TITLE
chore: http header with metrics

### PR DESCRIPTION
**Pull request by shuiyisong**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

This pr is a follow-up of https://github.com/GreptimeTeam/greptimedb/pull/3466. 
Now that we have `OutputMeta`, we can easily bring metrics from execution level up to the HTTP interface and put them into HTTP header. 
This pr covers basically every HTTP entry with read or write request, returning a HTTP header indicating the cost of the request.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

File: src/servers/src/http/prometheus.rs [hunk 1]
```diff
@@ -39,9 +39,11 @@ use query::parser::{PromQuery, DEFAULT_LOOKBACK_STRING};
 use schemars::JsonSchema;
 use serde::de::{self, MapAccess, Visitor};
 use serde::{Deserialize, Serialize};
+use serde_json::Value;
 use session::context::QueryContextRef;
 use snafu::{Location, ResultExt};
 
+use super::header::collect_plan_metrics;
```
Comment (waynexia) [inline src/servers/src/http/prometheus.rs:46]: style: no `super` path outside test mod

File: src/operator/src/statement/copy_table_from.rs [hunk 2]
```diff
@@ -419,28 +422,40 @@ impl StatementExecutor {
...
     }
 }
 
 /// Executes all pending inserts all at once, drain pending requests and reset pending bytes.
 async fn batch_insert(
-    pending: &mut Vec<impl Future<Output = Result<usize>>>,
+    pending: &mut Vec<impl Future<Output = Result<Output>>>,
     pending_bytes: &mut usize,
-) -> Result<usize> {
+) -> Result<(usize, usize)> {
```
Comment (waynexia) [inline src/operator/src/statement/copy_table_from.rs:449]: Prefer to document these two return values

**Review by waynexia — APPROVED** [review]
LGTM 🚀

**Review by v0y4g3r — APPROVED** [review]
LGTM

**Comment by tisonkun** [comment]
This seems a `refactor` rather than a `chore` ... What a big change set 🤣

**Comment by sunng87** [comment]
This is actually a feature :joy_cat:

**Comment by tisonkun** [comment]
Although we miss the chance to correct the commit message, let's change the PR title ...

**Pull request merged**